I was testing this patch and got while trying to open a workflow:
```
10.0.3.192 - - [09/Jan/2021:00:13:19 +0000] "GET /api/datatypes/types_and_mapping HTTP/1.1" 200 - "https://galaxy.aws.islandcompare.ca/workflow/editor?id=93a9adef3e10792c" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/87.0.4280.88 Safari/537.36"
[pid: 32|app: 0|req: 800/800] 10.0.3.192 () {50 vars in 958 bytes} [Sat Jan  9 00:13:19 2021] GET /api/datatypes/types_and_mapping => generated 107897 bytes in 22 msecs (HTTP/1.1 200) 3 headers in 139 bytes (1 switches on core 1)
galaxy.workflow.modules INFO 2021-01-09 00:13:19,193 [p:32,w:1,m:0] [uWSGIWorker1Core3] parameter_def [{'parameter_type': 'text', 'optional': False}], how [none]
galaxy.workflow.modules INFO 2021-01-09 00:13:19,194 [p:32,w:1,m:0] [uWSGIWorker1Core3] parameter_def [{'default': 'False', 'restrictions': ['True', 'False'], 'parameter_type': 'text', 'optional': True}], how [staticRestrictions]
galaxy.workflow.modules INFO 2021-01-09 00:13:19,195 [p:32,w:1,m:0] [uWSGIWorker1Core3] parameter_def [{'default': 'False', 'restrictions': ['True', 'False'], 'parameter_type': 'text', 'optional': True}], how [staticRestrictions]
galaxy.workflow.modules INFO 2021-01-09 00:13:19,196 [p:32,w:1,m:0] [uWSGIWorker1Core3] parameter_def [{'parameter_type': 'text', 'optional': False}], how [none]
galaxy.workflow.modules INFO 2021-01-09 00:13:19,199 [p:32,w:1,m:0] [uWSGIWorker1Core3] parameter_def [{'default': '', 'parameter_type': 'text', 'optional': True}], how [none]
galaxy.workflow.modules INFO 2021-01-09 00:13:19,200 [p:32,w:1,m:0] [uWSGIWorker1Core3] parameter_def [{'default': '', 'parameter_type': 'text', 'optional': True}], how [none]
127.0.0.1 - - [09/Jan/2021:00:13:19 +0000] "GET /api/version HTTP/1.1" 200 - "-" "-"
[pid: 32|app: 0|req: 802/801] 127.0.0.1 () {16 vars in 186 bytes} [Sat Jan  9 00:13:19 2021] GET /api/version => generated 39 bytes in 10 msecs (HTTP/1.1 200) 3 headers in 139 bytes (1 switches on core 0)
galaxy.workflow.modules INFO 2021-01-09 00:13:19,214 [p:32,w:1,m:0] [uWSGIWorker1Core3] parameter_def [{'parameter_type': 'text', 'optional': False}], how [none]
galaxy.workflow.modules INFO 2021-01-09 00:13:19,346 [p:32,w:1,m:0] [uWSGIWorker1Core3] parameter_def [{'parameter_type': 'text', 'optional': False}], how [none]
galaxy.workflow.modules INFO 2021-01-09 00:13:19,347 [p:32,w:1,m:0] [uWSGIWorker1Core3] parameter_def [{'parameter_type': 'text', 'optional': True}], how [none]
galaxy.workflow.modules INFO 2021-01-09 00:13:19,353 [p:32,w:1,m:0] [uWSGIWorker1Core3] parameter_def [{'parameter_type': 'text', 'optional': True}], how [none]
10.0.3.192 - - [09/Jan/2021:00:13:19 +0000] "GET /workflow/load_workflow?_=true&id=93a9adef3e10792c&version=0 HTTP/1.1" 500 - "https://galaxy.aws.islandcompare.ca/workflow/editor?id=93a9adef3e10792c" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/87.0.4280.88 Safari/537.36"
Traceback (most recent call last):
  File "/srv/galaxy/lib/galaxy/web/framework/middleware/error.py", line 153, in __call__
    app_iter = self.application(environ, sr_checker)
  File "/srv/galaxy/venv/lib/python3.8/site-packages/paste/recursive.py", line 85, in __call__
    return self.application(environ, start_response)
  File "/srv/galaxy/venv/lib/python3.8/site-packages/paste/httpexceptions.py", line 640, in __call__
    return self.application(environ, start_response)
  File "/srv/galaxy/lib/galaxy/web/framework/base.py", line 141, in __call__
    return self.handle_request(environ, start_response)
  File "/srv/galaxy/lib/galaxy/web/framework/base.py", line 220, in handle_request
    body = method(trans, **kwargs)
  File "/srv/galaxy/lib/galaxy/web/framework/decorators.py", line 56, in call_and_format
    rval = func(self, trans, *args, **kwargs)
  File "/srv/galaxy/lib/galaxy/webapps/galaxy/controllers/workflow.py", line 732, in load_workflow
    return workflow_contents_manager.workflow_to_dict(trans, stored, style="editor", version=version)
  File "/srv/galaxy/lib/galaxy/managers/workflows.py", line 505, in workflow_to_dict
    wf_dict = self._workflow_to_dict_editor(trans, stored, workflow)
  File "/srv/galaxy/lib/galaxy/managers/workflows.py", line 769, in _workflow_to_dict_editor
    'outputs': module.get_all_outputs(),
  File "/srv/galaxy/lib/galaxy/workflow/modules.py", line 478, in get_all_outputs
    subworkflow_dict = workflow_contents_manager._workflow_to_dict_editor(trans=self.trans,
  File "/srv/galaxy/lib/galaxy/managers/workflows.py", line 748, in _workflow_to_dict_editor
    upgrade_message = module.check_and_update_state()
  File "/srv/galaxy/lib/galaxy/workflow/modules.py", line 1501, in check_and_update_state
    return self.tool.check_and_update_param_values(self.state.inputs, self.trans, workflow_building_mode=True)
  File "/srv/galaxy/lib/galaxy/tools/__init__.py", line 1786, in check_and_update_param_values
    visit_input_values(self.inputs, values, validate_inputs)
  File "/srv/galaxy/lib/galaxy/tools/parameters/__init__.py", line 164, in visit_input_values
    visit_input_values(input.cases[values['__current_case__']].inputs, values, callback, new_name_prefix, label_prefix, parent_prefix=name_prefix, **payload)
  File "/srv/galaxy/lib/galaxy/tools/parameters/__init__.py", line 170, in visit_input_values
    callback_helper(input, input_values, name_prefix, label_prefix, parent_prefix=parent_prefix, context=context)
  File "/srv/galaxy/lib/galaxy/tools/parameters/__init__.py", line 131, in callback_helper
    new_value = callback(**args)
  File "/srv/galaxy/lib/galaxy/tools/__init__.py", line 1772, in validate_inputs
    value, error = check_param(request_context, input, value, context)
  File "/srv/galaxy/lib/galaxy/tools/parameters/__init__.py", line 187, in check_param
    value = param.from_json(value, trans, param_values)
  File "/srv/galaxy/lib/galaxy/tools/parameters/basic.py", line 1240, in from_json
    return super().from_json(value, trans, other_values)
  File "/srv/galaxy/lib/galaxy/tools/parameters/basic.py", line 880, in from_json
    legal_values = self.get_legal_values(trans, other_values)
  File "/srv/galaxy/lib/galaxy/tools/parameters/basic.py", line 1331, in get_legal_values
    if value is not None and value not in legal_values and self.is_file_empty(trans, other_values):
  File "/srv/galaxy/lib/galaxy/tools/parameters/basic.py", line 1341, in is_file_empty
    if not dataset.has_data():
AttributeError: 'ConnectedValue' object has no attribute 'has_data'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/srv/galaxy/lib/galaxy/web/framework/middleware/batch.py", line 80, in __call__
    return self.application(environ, start_response)
  File "/srv/galaxy/lib/galaxy/web/framework/middleware/request_id.py", line 15, in __call__
    return self.app(environ, start_response)
  File "/srv/galaxy/lib/galaxy/web/framework/middleware/xforwardedhost.py", line 23, in __call__
    return self.app(environ, start_response)
  File "/srv/galaxy/lib/galaxy/web/framework/middleware/translogger.py", line 70, in __call__
    return self.application(environ, replacement_start_response)
  File "/srv/galaxy/lib/galaxy/web/framework/middleware/error.py", line 161, in __call__
    start_response('500 Internal Server Error',
  File "/srv/galaxy/lib/galaxy/web/framework/middleware/translogger.py", line 69, in replacement_start_response
    return start_response(status, headers, exc_info)
SystemError: <built-in function uwsgi_spit> returned a result with an error set
[pid: 32|app: 0|req: 802/802] 10.0.3.192 () {50 vars in 1013 bytes} [Sat Jan  9 00:13:19 2021] GET /workflow/load_workflow?_=true&id=93a9adef3e10792c&version=0 => generated 0 bytes in 1164 msecs (HTTP/1.1 500) 1 headers in 63 bytes (0 switches on core 3)
127.0.0.1 - - [09/Jan/2021:00:13:21 +0000] "GET /api/version HTTP/1.1" 200 - "-" "-"
```

I believe this is related/

--- a/lib/galaxy/tools/parameters/basic.py
+++ b/lib/galaxy/tools/parameters/basic.py
@@ -1349,7 +1349,7 @@ class ColumnListParameter(SelectToolParameter):
             # Use representative dataset if a dataset collection is parsed
             if isinstance(dataset, trans.app.model.HistoryDatasetCollectionAssociation):
                 dataset = dataset.to_hda_representative()
-            if not dataset.has_data():
+            if not is_runtime_value(dataset) and not dataset.has_data():
                 return True
         return False
 

--- a/lib/galaxy/tools/parameters/basic.py
+++ b/lib/galaxy/tools/parameters/basic.py
@@ -1349,7 +1349,7 @@ class ColumnListParameter(SelectToolParameter):
             # Use representative dataset if a dataset collection is parsed
             if isinstance(dataset, trans.app.model.HistoryDatasetCollectionAssociation):
                 dataset = dataset.to_hda_representative()
-            if not is_runtime_value(dataset) and not dataset.has_data():
+            if is_runtime_value(dataset) or not dataset.has_data():
                 return True
         return False
 

